core/src/ModelArray.cpp

```diff
@@ -200,9 +200,23 @@ void ModelArray::setNComponents(std::map<Type, size_t> cMap)
     }
 }
 
-void ModelArray::setDimension(Dimension dim, size_t length)
-{
-    definedDimensions.at(dim).length = length;
+#ifdef USE_MPI
+void ModelArray::setDimension(
+    Dimension dim, size_t global_length, size_t local_length, size_t start)
+#else
+void ModelArray::setDimension(Dimension dim, size_t global_length)
+#endif
+{
+#ifdef USE_MPI
+    definedDimensions.at(dim).global_length = global_length;
+    definedDimensions.at(dim).local_length = local_length;
+    definedDimensions.at(dim).start = start;
+#else
+    // if MPI is not used then set the local_length to be the same as the global
+    definedDimensions.at(dim).global_length = global_length;
+    definedDimensions.at(dim).local_length = global_length;
+    definedDimensions.at(dim).start = 0;
+#endif
```

Comment:
Rewrite this to only require one `#ifdef`

Reply:
done